src/scripts/Selector/SelectorDocument.js

```diff
@@ -27,80 +26,65 @@ export default class SelectorDocument extends Selector {
 		return false;
 	}
 
-	_getData(parentElement) {
-		var elements = this.getDataElements(parentElement);
+	downloadsAttachments() {
+		return !!this.downloadDocument;
+	}
 
-		var dfd = $.Deferred();
+	async _getData(parentElement) {
+		const elements = this.getDataElements(parentElement).filter(element => 'href' in element);
+		const urls = elements.map(element =>
+			this.stringReplace(element.href, this.stringReplacement)
+		);
 
-		// return empty record if not multiple type and no elements found
-		if (this.multiple === false && elements.length === 0) {
-			var data = {};
-			data[this.id] = null;
-			dfd.resolve([data]);
-			return dfd;
+		const result = {};
+		if (this.multiple) {
```

Comment:
Looks like ternary operator. But maybe 2 ternary operators is too much for one expression.

Reply:
It does, but eslint disallows nested ternaries, so I split them